### DISCUSSION TITLE
Is zram0 governed by a file system?

### DISCUSSION THREAD

**Discussion by Ricky-Tigg** [util-linux/util-linux#3193] [Q&A] [answered]
Hello. Here the zram0 partition shows no file system associated with it.
```
$ lsblk -o name,maj:min,fstype,fsver,label,mountpoint,pttype /dev/zram0 
NAME  MAJ:MIN FSTYPE FSVER LABEL MOUNTPOINT PTTYPE
zram0 252:0                      [SWAP]     
```
Here it is not referred to as a file system.
```
$ sudo file -bs /dev/zram0 
Linux swap file, 4k page size, little endian, version 1, size ... pages, ... bad pages, LABEL=zram0, UUID=...
```
So should we deduce that zram0 is not governed by a file system? Or could this be an omission in lsblk, therefore the file system would be _Linux swap file_ in _version 1_?

**Comment by karelzak** [comment, marked as answer]
By default, lsblk reads data from udevd. However, udevd does not collect file system-related information from zrams. This can be seen in the output of `udevadm info --export /dev/zram0`, where there are no ID_FS_* variables present.
 
According to the rules in https://github.com/systemd/systemd/blob/main/rules.d/60-persistent-storage.rules.in, zram is not included in the list of allowed devices for KERNEL!=.
 
In the current git tree, lsblk can be forced to read data directly from the device using libblkid. In this case, all properties are retrieved.
```
sudo ./lsblk --fs --properties-by blkid /dev/zram0
NAME  FSTYPE FSVER LABEL UUID                                 FSAVAIL FSUSE% MOUNTPOINTS
zram0 swap               e13fc0c2-1660-47b6-a013-ef3102175fcc                [SWAP]
```

We need to report it to systemd (udev) guys.

**Comment by karelzak** [comment]
Sent systemd PR: https://github.com/systemd/systemd/pull/34361
